torch/csrc/jit/python/init.cpp

```diff
@@ -122,6 +125,24 @@ using ::c10::FunctionSchema;
 using caffe2::serialize::PyTorchStreamReader;
 using caffe2::serialize::PyTorchStreamWriter;
 
+class PythonSymbolicIntNode: public c10::SymbolicIntNode {
+public:
+  PythonSymbolicIntNode(py::object pyobj): 
+    c10::SymbolicIntNode() {
+      pyobj_ = std::make_shared<c10::SafePyObject>(pyobj.release().ptr(), getPyInterpreter());
+    };
+
+  virtual SymbolicIntNode* add(SymbolicIntNode* other) override {
+    auto pother = dynamic_cast<PythonSymbolicIntNode*>(other);
+    py::gil_scoped_acquire acquire;
+    auto r = getPyObj().attr("__add__")(pother->getPyObj());
```

Comment:
The code here pretends that the Python objects here can be independent, but not enough infrastructure really is in place to actually support multiple symints with different backing Python objects floating around. We know how to do this properly (copy the Tensor subclass design), but if we're not really going to do it properly we might as well call a spade a spade and just have some global object that we call into to reflect Python calls back. At least it's simple and doesn't give people any ideas that something more general might work.